zebra/zebra_rib.c

```diff
@@ -1042,7 +1049,11 @@ int rib_install_kernel(struct route_node *rn, struct route_entry *re,
 			else
 				UNSET_FLAG(nexthop->flags, NEXTHOP_FLAG_FIB);
 		}
-	}
+		zsend_route_notify_owner(re->type, re->vrf_id,
+					 p, ZAPI_ROUTE_INSTALLED);
+	} else
+		zsend_route_notify_owner(re->type, re->vrf_id,
+					 p, ZAPI_ROUTE_FAIL_INSTALL);
```

Comment:
Looks good to me but we'll probably need to revisit this once we finish the new southbound interface. For example, if the route was successfully installed in the fast-path and not in the kernel (or vice-versa), should we notify success or failure? Something to keep in mind for later.

Reply:
yes totally agree we will need to think this through carefully and it's on my radar.